src/torrent.cpp

```diff
@@ -4102,24 +4104,20 @@ bool is_downloading_state(int const st)
 		if (m_picker)
 			m_picker->get_downloaders(downloaders, index);
 
-		// if we know which blocks failed null out all the non-failing
-		// downloaders
-		if (!blocks.empty() && !downloaders.empty())
-		{
-			auto begin = downloaders.begin();
-			for (int const block : blocks)
-			{
-				std::fill(begin, downloaders.begin() + block, nullptr);
-				begin = downloaders.begin() + block + 1;
-			}
-			std::fill(begin, downloaders.end(), nullptr);
-		}
-
```

Comment:
it took me a few passes to understand what this was doing. I think my change simplifies it a bit by combining this filter with copying the peers into the set below